### DISCUSSION TITLE
Is it possible to get the default branch from a repository?

### DISCUSSION THREAD

**Discussion by yvsean** [microsoft/PowerShellForGitHub#344] [General]
Hi,

I'd like to be able to fetch the default branch for a repository. I can do this no problem with the GitHub API directly, but using Get-GitHubRepository (which I presume is the analog), the only data that gets output is the visibility flag for the repo. Is it possible to get the detailed information the GitHub API produces, particularly the default branch?

**Comment by HowardWolosky** [comment]
That information is already available.  It's simply the difference between the "view" of the information (how PowerShell renders a known object at the command-line) and the object itself.

`Get-GitHubRepository` returns back a `GitHub.Repository` object, which happens to have a view that displays a limited number of properties.  If you want the default branch, it's still there, it's just not displayed by default.

Any of these will work:

```powershell
(Get-GitHubRepository -OwnerName microsoft -RepositoryName PowerShellForGitHub).default_branch
```

or

```powershell
$repo = Get-GitHubRepository -OwnerName microsoft -RepositoryName PowerShellForGitHub
$repo.default_branch
```

You can see all of the properties by piping it into Get-Member:

```powershell
Get-GitHubRepository -OwnerName microsoft -RepositoryName PowerShellForGitHub | gm
```

**Reply by yvsean** [reply]
Well... duh. Still relatively new to PowerShell. Thank you. :)
